src/test/kotlin/com/atlassian/performance/tools/virtualusers/ExploratoryVirtualUserTest.kt

```diff
@@ -43,6 +45,30 @@ class ExploratoryVirtualUserTest {
         assertLoadInRange(actualLoad, closeToMaxLoad, maxLoad)
     }
 
+    @Test
+    fun shouldNotRetryOnLoginAction() {
+        val server = QuickServer()
+        val logInAction = object : Action {
+            override fun run() {
+                throw Exception("Failed login attempt.")
+            }
+        }
+        val virtualUser = prepareVu(
+            actions = listOf(server),
+            maxLoad = TemporalRate(1_000_000.00, ofSeconds(1)),
+            logInAction = logInAction
+        )
+
+        val done = AutoCloseableExecutorService(Executors.newSingleThreadExecutor()).use { executorService ->
```

Comment:
Does it still repro when we switch this to `applyLoad(virtualUser, ofSeconds(1)`?
Instead of a bool we'd implicitly assert on exception being thrown or not.

Reply:
> Instead of a bool we'd implicitly assert on exception being thrown or not.

For the current implementation, it'll catch the exception and retry.

Reply:
Ah, we'd need to interrupt it like `LoadTest` does. Currently the local `applyLoad` only simulates the `ExploratoryVirtualUser.shutdown.set(true)`, but not the interrupt. Fair enough.